Documentation/Proposals/NNNN-public-issues-api.md

```diff
@@ -0,0 +1,145 @@
+`Issue.record()` instance method to report that Issue. This provides a clean,
+easy to maintain way to create and record custom Issues that assertion tools
+are already used to because it is analogous to the XCTIssue API in XCTest.
+
+## Detailed design
+
+Create a public initializer for `Issue`, which is not gated by
+`@_spi(ForToolsIntegrationOnly)`:
+
+```swift
+public struct Issue: Sendable {
+  // ...
+
+  /// Initialize an issue instance with the specified details.
+  ///
+  /// - Parameters:
+  ///   - kind: The kind of issue this value represents.
+  ///   - comments: An array of comments describing the issue. This array may be
+  ///     empty.
+  ///   - sourceContext: A ``SourceContext`` indicating where and how this issue
+  ///     occurred. This defaults to a default source context returned by
+  ///     calling ``SourceContext/init(backtrace:sourceLocation:)`` with zero
+  ///     arguments.
+  public init(
+    kind: Kind,
+    comments: [Comment],
+    sourceContext: SourceContext = .init()
+  ) {
+    self.kind = kind
+    self.comments = comments
+    self.sourceContext = sourceContext
+  }
+
+  // ...
+}
+```
+
+Additionally, create a public `record()` method on Issue, which takes no
+arguments, and simply calls the `Issue.record(configuration:)` instance
+method with nil:
+
+```swift
+extension Issue: {
+  /// Record this issue by wrapping it in an ``Event`` and passing it to the
+  /// current event handler.
+  ///
+  /// - Returns: The issue that was recorded (`self` or a modified copy of it.)
+  @discardableResult
+  public func record() -> Self {
+    record(configuration: nil)
+  }
+}
+```
+
+## Source compatibility
+
+This is strictly an additive change. No deletion of code, nothing is being made
+private. Only new code and making existing code public.
+
+## Integration with supporting tools
+
+Third-party assertion tools would be able to directly create an `Issue`
+and then report it using the `Issue.report()` instance method. `Issue.report()`
+would then work similarly to how it does now. This flow is analogous to
+reporting an issue in XCTest using the XCTIssue API.
+
+## Future directions
+
+Also necessary for supporting third-party assertion tools is providing a stable
+and reliable API for accessing the current test. This somewhat exists with the
+current `Test.current` static property, but that returns the
+semantically-incorrect value of nil if you access it from a detached task. This
+will be defined in a future proposal.
+
+A future direction is to remove all of the `@_spi(ForToolsIntegrationOnly)`
+annotations. This would allow third party tools to integrate with Swift
+Testing regardless of the distribution method they use.
+This was left out to keep this proposal small & focused, but the same rationale
+behind making the `Issue` initializer public and available everywhere applies
+to removing the `@_spi(ForToolsIntegrationOnly)` annotation from everything.
```

Comment:
I don't think it's actually a goal per se to eventually remove _every_ usage of `@_spi(ForToolsIntegrationOnly)`. There may be things that users of this library may never need, but tools always need. APIs marked `@_spi(Experimental)`, however, are more likely to be considered for promotion to public API. There's more context and background about this in `SPI.md`(https://github.com/apple/swift-testing/blob/main/Documentation/SPI.md).